include/bluetooth/bluetooth.h

```diff
@@ -398,6 +463,118 @@ int bt_le_adv_update_data(const struct bt_data *ad, size_t ad_len,
  */
 int bt_le_adv_stop(void);
 
+/** @brief Increment an advertising set`s reference count.
+ *
+ *  Increment the reference count of an advertising set object.
+ *
+ *  @param adv Advertising set object.
+ *
+ *  @return Advertising set object with incremented reference count.
+ */
+struct bt_adv *bt_adv_ref(struct bt_adv *adv);
+
+/** @brief Decrement an advertising set's reference count.
+ *
+ *  Decrement the reference count of an advertising setobject.
+ *
+ *  @param adv Advertising set object.
+ */
+void bt_adv_unref(struct bt_adv *adv);
+
+/** @brief Create advertising set.
+ *
+ *  Create a new advertising set and set advertising parameters.
+ *
+ *  @param param Advertising parameters.
```

Comment:
@jhedberg @trond-snekvik I'm wondering if changing advertising parameters of the advertising set is needed.
Is this something that would be needed for mesh for example?

At the moment the advertising set has to be deleted and created again to change the advertising parameters.
Not all of the advertising parameters can be changed without creating a new advertising set. For example we cannot switch to legacy PDUs when advertising data currently exceeds 31 bytes.